### DISCUSSION TITLE
Use `defer` for GA script instead of `async`

### DISCUSSION THREAD

**Discussion by PiotrGrobelak** [johannschopplich/nuxt-gtag#10] [Q&A] [answered]
How can I load the nuxt-gtag library after the DOM content has loaded in Nuxt 3? I have tried adding the defer property to the configuration object passed to nuxt-gtag, but I am getting an error saying that client is not a known property. Is there a way to achieve this using the defineNuxtPlugin function? Thank you for any help or suggestions you can provide!

**Comment by johannschopplich** [comment, marked as answer]
Hey there, Google's `gtag.js` library is added `async` to the document's `<head>` by default, which mimics Google's snippet:

https://github.com/johannschopplich/nuxt-gtag/blob/008020e5030b96d404215b694240e814c23f9abb/src/runtime/plugin.client.ts#L24

Do you prefer `defer` as a separate option to `defer``

**Reply by PiotrGrobelak** [reply]
Yeah, it would be nice to have options to choose from because some websites require faster loading times to improve their score in Lighthouse and provide a better user experience.

**Reply by johannschopplich** [reply]
I see, good idea. I have added the `loadingStrategy` config option in https://github.com/johannschopplich/nuxt-gtag/commit/ae590f43c3ecace62aa0f0f11163ac4d071ad0a6. The default is now `defer` instead of `async`.

It's available in v0.4.0+.

Usage:

```ts
// `nuxt.config.ts`
export default defineNuxtConfig({
  modules: ['nuxt-gtag'],

  gtag: {
    id: 'G-XXXXXXXXXX',
    loadingStrategy: 'defer'
  }
})
```

**Reply by PiotrGrobelak** [reply]
Thanks for the quick implementation and release :) I will check the configuration and results soon.
